code/src/hooks/use-infinite-search/useInfiniteSearch.ts

```diff
@@ -0,0 +1,75 @@
+import { skipToken } from "@reduxjs/toolkit/query/react";
+import { useEffect, useRef, useState } from "react";
+
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useDebouncedValue from "@/hooks/use-debounced-value/useDebouncedValue";
+import { useGetUserProductsBySearchQuery } from "@/store/api/productsApi";
+import { GetUserProductsBySearchQueryResponse } from "@/types/product.types";
+
+const RESULTS_PER_PAGE = 10;
+const MIN_QUERY_LENGTH = 4;
+const DEBOUNCE_DELAY_MS = 500;
+
+const useInfiniteSearch = (query: string) => {
```

Comment:
I suggest creating optional second argument and allow future developers to configure results per page, min query length and delay in ms. You can set default values same as you declared

Reply:
Created